homeworks/homework_1/task_1/matrix.py

```diff
@@ -0,0 +1,51 @@
+from typing import List
+
+from homeworks.homework_1.task_1.vector import Vector
+
+
+class Matrix:
+    __matrix: List[List[float]]
+    __rows: int
+    __columns: int
+
+    def __init__(self, matrix: List[List[float]], eps=0.01):
+        if len(matrix) != 0:
+            row_length = len(matrix[0])
+            for row in matrix:
+                if len(row) != row_length:
+                    raise ValueError("Rows in a matrix should be equal length")
+        self.eps = eps
+        self.__rows = len(matrix)
+        self.__columns = len(matrix[0])
+        self.__matrix = matrix
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Matrix):
+            return NotImplemented
+
+        for i in range(self.__rows):
+            for j in range(self.__columns):
+                if (self.__matrix[i][j] - other.__matrix[i][j]) > self.eps:
+                    return False
+        return True
```

Comment:
For the purpose of this homework, let's rewrite this as a single statement (using `for` loops after the statement) and using keywords `all` or `any`.